.github/workflows/check-compatibility.yml

```diff
@@ -7,9 +7,14 @@ on:
 jobs:
   build:
     if: github.repository == 'opensearch-project/OpenSearch'
+    permissions:
+      contents: read # to fetch code (actions/checkout)
+      pull-requests: write # to create or update comment (peter-evans/create-or-update-comment)
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
This changes the environment of the GitHub action to get code from a pull request and then run `./gradlew checkCompatibility` with write permissions on pull requests.  This seems like it opens up a security vulnerability in the GitHub action.

I think we need a separate workflow that is triggered on the pull request that doesn't have access to the write permissions, we could have it output its results and then pull it in like how the gradlew-check.out is used

Reply:
I think that was gradle check as it as well here https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/gradle-check.yml#L16-L20

Reply:
I added another job with write permission to only comment on the PR. Please see the latest commit for the changes. Thank you!